Add image api to flask server

jbewing opened the pull request:
This PR adds three new endpoints for the Image API to the Flask server:
- `GET /api/images` which returns a list of all images stored in the DB with the following body:
```json
[
  {
     "href": "/api/images/{id}", 
     "title": "<imageTitle>" # title that the image is uploaded with
  }
...
]
```
- `GET /api/images/{id}` -> returns the image itself
- `POST /api/images` -> accepts file upload input and returns a 201 with the following body:
```json
{
  "status": "created",
   "href": "/api/images/<imageId>"
}
```

_Note:_ the file format isn't necessarily restrictive to images at this point. Tbh, you could just use this as a BLOB storage like S3.

#### Testing
```bash
# I've run the following
poetry run flask run
curl -F "image=@/Users/jbewing/Desktop/a_golden_retriever.jpeg" 127.0.0.1:5000/api/images
```
And verified that [http://127.0.0.1:5000/api/images](http://127.0.0.1:5000/api/images) returns 
```json
[{"href":"/api/images/1","title":"a_golden_retriever.jpeg"}]
```

and [http://127.0.0.1:5000/api/images/1](http://127.0.0.1:5000/api/images/1) returns 
<img width="1264" alt="Screen Shot 2020-10-16 at 2 30 27 AM" src="https://user-images.githubusercontent.com/13725922/96220881-974e6200-0f57-11eb-993b-21a4a5d3bb15.png">

*Related Issues*: https://github.com/brodigan-e/capstone-POV/issues/4 https://github.com/brodigan-e/capstone-POV/issues/5
cc @DarkAce65

jbewing committed:
Add image api to flask server

jbewing committed:
Update image api to save images with uuids

jbewing committed:
Update creation endpoint to return path to created image

jbewing committed:
Update image api to use href instead of id

href is probably a bit cleaner for external responses.

DarkAce65 commented on server/.flaskenv:
```diff
@@ -0,0 +1 @@
+FLASK_APP=server/app.py
```
This can just go in a `.env` file - we've got python-dotenv loaded so environment variables set there will automatically be set

DarkAce65 commented on server/README.md:
```diff
@@ -4,4 +4,16 @@
 
 - `poetry install` to download all required dependencies and set up a virtual environment
-- `poetry run task run` to start the Django server
+- `poetry run flask run` to start the Flask server
```
`poetry run task run` runs the `run` task in the pypoetry.toml file (under `[tool.taskipy.tasks]`) so you can just add more tasks there for db migration (also, the `run` task is already configured for `flask run`)

_phew, lots of runs and tasks in that sentence_ 😝

jbewing replied:
I added a couple of new _tasks_:
- `add-migration` => `flask db migrate` (looks like command line args get passed from the `taskipy` docs)
- `apply-migrations` => `flask db upgrade` (debated making this a pre_task of the "run task", but I do like that a schema migration needs to be an explicit "task"—sorry to use this overused word—that is run)

jbewing replied:
Not going to add `revert-migrations` or anything like that because 🤞 hopefully we don't ever need to use them 😄

jbewing committed:
PR feedback

- Move my .env stuff into an actual .env file (oopsies)
- Add poetry tasks for db migration stuff + update docs
